lib/reek/source/source_locator.rb

```diff
@@ -8,64 +8,27 @@ module Source
     # Finds Ruby source files in a filesystem.
     #
     class SourceLocator
+      include Enumerable
+      extend Forwardable
+
+      def_delegator :sources, :each
```

Comment:
And this is my biggest problem with proposed changes. A lot of implicitness here and it's partially obscure.
I can see why it makes sense that a SourcePath is enumerable, but making SourceLocator enumerable seems over-engineered.
I'd rather drop that.